client/dashboard/app/root/index.tsx

```diff
@@ -19,15 +19,35 @@ const WebpackBuildMonitor = lazy(
 );
 
 function Root() {
-	const { LoadingLogo = WordPressLogo } = useAppContext();
+	const { name, LoadingLogo = WordPressLogo } = useAppContext();
 	const isFetching = useIsFetching();
 	const router = useRouterState();
+	const routeMeta = useRouterState( {
+		select: ( state ) => {
+			return state.matches.map( ( match ) => match.meta! ).filter( Boolean );
+		},
+	} );
+
 	const isNavigating = router.status === 'pending';
 	// A little trick after investigation router state: it will initially be
 	// empty, but remain set after subsequent navigations.
 	// https://tanstack.com/router/latest/docs/framework/react/api/router/RouterStateType#resolvedlocation-property
 	const isInitialLoad = ! router.resolvedLocation;
 
+	const title = useMemo( () => {
+		return routeMeta
+			.map( ( metas ) => metas.find( ( meta ) => meta?.title )?.title )
+			.filter( Boolean )
+			.reverse()
+			.join( ' ‹ ' );
+	}, [ routeMeta ] );
+
+	useEffect( () => {
+		if ( title ) {
+			document.title = `${ title } - ${ name }`;
```

Comment:
I think we're using en dash across WP (and .org as well).
```suggestion
			document.title = `${ title } – ${ name }`;
```